include/sdp/nrfe_common.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRFE_COMMON_H__
+#define NRFE_COMMON_H__
+
+#include <stdatomic.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct __packed {
+	atomic_bool locked;
```

Comment:
Use the zephyr atomic primitive in #include <zephyr/sys/atomic.h>

Additionally, don't typedefine structs:
```
struct nrfe_shared_data_lock {
...
} __packed;
```